engine/sawtooth_poet_engine/engine.py

```diff
@@ -219,17 +221,24 @@ def _handle_new_block(self, block):
         if self._check_consensus(block):
```

Comment:
I think this check needs to be moved to _handle_valid_block()

Reply:
Added a commit for this